includes/Modules/Tag_Manager.php

```diff
@@ -566,38 +554,31 @@ protected function create_data_request( Data_Request $data ) {
 	 * Creates GTM Container.
 	 *
 	 * @since 1.0.0
-	 * @param string       $account_id The account ID.
+	 * @param string       $account_id    The account ID.
 	 * @param string|array $usage_context The container usage context(s).
 	 *
-	 * @return mixed Container ID on success, or WP_Error on failure.
+	 * @return string Container public ID.
 	 */
 	protected function create_container( $account_id, $usage_context = self::USAGE_CONTEXT_WEB ) {
 		$restore_defer = $this->with_client_defer( false );
 
 		// Use site name for container, fallback to domain of reference URL.
 		$container_name = get_bloginfo( 'name' ) ?: wp_parse_url( $this->context->get_reference_site_url(), PHP_URL_HOST );
+		// Prevent naming conflict (Tag Manager does not allow more than one with same name).
+		if ( self::USAGE_CONTEXT_AMP === $usage_context ) {
+			$container_name .= ' AMP';
+		}
 		$container_name = self::sanitize_container_name( $container_name );
 
 		$container = new Google_Service_TagManager_Container();
 		$container->setName( $container_name );
 		$container->setUsageContext( (array) $usage_context );
 
-		try {
-			$container = $this->get_tagmanager_service()->accounts_containers->create( "accounts/{$account_id}", $container );
-		} catch ( Google_Service_Exception $e ) {
-			$restore_defer();
-			$message = $e->getErrors();
-			if ( isset( $message[0]['message'] ) ) {
-				$message = $message[0]['message'];
-			}
-			return new WP_Error( $e->getCode(), $message );
-		} catch ( Exception $e ) {
-			$restore_defer();
-			return new WP_Error( $e->getCode(), $e->getMessage() );
-		}
```

Comment:
I'm not sure we should remove this try-catch from here. With the updated code, we don't `$restore_defer()` any more when an exception happens, which could become problematic. We should keep the try-catch to do that, we could just re-throw the exception.

Reply:
Good catch! Updated 👍